src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -51,710 +78,284 @@ public class AnalysisService : IDisposable
-            try
-            {
-                if (s_existingRequestCancellation != null)
-                {
-                    // Try to cancel the request
-                    s_existingRequestCancellation.Cancel();
-
-                    // If cancellation didn't throw an exception,
-                    // clean up the existing token
-                    s_existingRequestCancellation.Dispose();
-                    s_existingRequestCancellation = null;
-                }
-            }
-            catch (Exception e)
-            {
-                // TODO: Catch a more specific exception!
-                _logger.LogError(
-                    string.Format(
-                        "Exception while canceling analysis task:\n\n{0}",
-                        e.ToString()));
-
-                TaskCompletionSource<bool> cancelTask = new TaskCompletionSource<bool>();
-                cancelTask.SetCanceled();
-                return Task.CompletedTask;
-            }
-
-            // If filesToAnalzye is empty, nothing to do so return early.
-            if (filesToAnalyze.Length == 0)
+            if (cancellationToken.IsCancellationRequested)
             {
-                return Task.CompletedTask;
+                return;
             }
 
-            // Create a fresh cancellation token and then start the task.
-            // We create this on a different TaskScheduler so that we
-            // don't block the main message loop thread.
-            // TODO: Is there a better way to do this?
-            s_existingRequestCancellation = new CancellationTokenSource();
-            bool scriptAnalysisEnabled = _configurationService.CurrentSettings.ScriptAnalysis.Enable ?? false;
-            return Task.Run(() => DelayThenInvokeDiagnosticsAsync(delayMilliseconds: 750, filesToAnalyze, scriptAnalysisEnabled, s_existingRequestCancellation.Token));
-        }
-
-        private async Task DelayThenInvokeDiagnosticsAsync(
-            int delayMilliseconds,
-            ScriptFile[] filesToAnalyze,
-            bool isScriptAnalysisEnabled,
-            CancellationToken cancellationToken)
-        {
-            // First of all, wait for the desired delay period before
-            // analyzing the provided list of files
             try
             {
-                await Task.Delay(delayMilliseconds, cancellationToken).ConfigureAwait(false);
+                await Task.Delay(_analysisDelayMillis, cancellationToken).ConfigureAwait(false);
             }
             catch (TaskCanceledException)
             {
-                // If the task is cancelled, exit directly
-                foreach (var script in filesToAnalyze)
-                {
-                    PublishScriptDiagnostics(
-                        script,
-                        script.DiagnosticMarkers);
-                }
+                // Ensure no stale markers are displayed
+                //foreach (ScriptFile script in filesToAnalyze)
+                //{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4961943)